crypto/zkp_bip340.c

```diff
@@ -32,6 +33,12 @@
 
 #include "zkp_bip340.h"
 
+// Initial hash value H for SHA-256 TapTweak:
+const uint32_t sha256_initial_taptweak_state[8] = {
+    0xd129a2f3UL, 0x701c655dUL, 0x6583b6c3UL, 0xb9419727UL,
+    0x95f4e232UL, 0x94fd54f4UL, 0xa2ae8d85UL, 0x47ca590bUL,
+};
+
```

Comment:
Does this optimisation bring notable improvement over using `sha256_Init` + 2x `sha256_Update`?

Reply:
I haven't done any measurements. The secp256k1-zkp library does this for all the tagged hashes, e.g. https://github.com/ElementsProject/secp256k1-zkp/blob/6b8733577e4a8e1a4f77aaa48691f188b0c42f5c/src/modules/schnorrsig/main_impl.h#L14-L44, so I did the same.

We don't use this optimization in sig_hasher.py for "TapSighash". That would require a little more work because of the Python layer, so I didn't want to optimize prematurely. But the case above was trivial to implement so I felt it worthwhile.

Reply:
Thanks for explanation. Since sec256k1-zkp lib does the same I am OK with keeping this. 👍